### PR TITLE
add shape as a distinct field. shape.npy etc

### PR DESCRIPTION
**Pull request by taillades**

### CODE REVIEW

File: rouskinhf/datapoint.py [hunk 1]
```diff
@@ -305,9 +313,12 @@ def from_json_line(ref, d, predict_structure=False, predict_dms=False):
         # create the datapoint
         sequence = d["sequence"]
         sequence = standardize_sequence(sequence)
+        shape = d["shape"] if "shape" in d else None
 
         if predict_structure and (not "structure" in d) and (not "paired_bases" in d):
-            d["structure"] = RNAstructure_singleton.predictStructure(sequence, dms=d["dms"] if "dms" in d else None)
+            if 'dms' in d and 'shape' in d:
+                print('DMS and SHAPE are both present in the json line. Using DMS.')
```
Comment (taillades) [inline rouskinhf/datapoint.py:317]: @AlbericDeLajarte we're using DMS by default if there are two data types in the dataset? or shall we just raise an error? It's an unlikely scenario anyways

File: rouskinhf/list_datapoints.py [hunk 2]
```diff
@@ -208,27 +227,29 @@ def drop_duplicates(df:pd.DataFrame, **kwargs):
             n_duplicates_datapoints = drop_duplicates(df, subset=['sequence','paired_bases'], inplace=True, ignore_index=True)
 
         # Keep only one datapoint per sequence and dms
-        if 'dms' in df.columns:
-            if not 'paired_bases' in df.columns: n_duplicates_datapoints = 0
-            n_duplicates_datapoints += drop_duplicates(df, subset=['sequence','dms'], inplace=True, ignore_index=True)
-        
+        for signal in ['dms', 'shape']:
```
Comment (taillades) [inline rouskinhf/list_datapoints.py:247]: @AlbericDeLajarte this whole filtering function needs review

File: rouskinhf/datapoint.py (hunk 1 above)
Comment (AlbericDeLajarte) [reply rouskinhf/datapoint.py:317]: Yeah for now we can raise an error, I guess all our dataset are separated into SHAPE or DMS

File: rouskinhf/list_datapoints.py [hunk 3]
```diff
@@ -208,27 +227,29 @@ def drop_duplicates(df:pd.DataFrame, **kwargs):
...
+            if signal in df.columns:
+                if not 'paired_bases' in df.columns: n_duplicates_datapoints = 0
+                n_duplicates_datapoints += drop_duplicates(df, subset=['sequence', signal], inplace=True, ignore_index=True)
+            
         # If there are multiple structures / dms with the same structure, keep none
-        n_same_seq_datapoints = drop_duplicates(df, subset=['sequence'], inplace=True, ignore_index=True, keep='first' if not ('paired_bases' in df.columns or 'dms' in df.columns) else False)
+        n_same_seq_datapoints = drop_duplicates(df, subset=['sequence'], inplace=True, ignore_index=True, keep='first' if not ('paired_bases' in df.columns or 'dms' in df.columns or 'shape' in df.columns) else False)
```
Comment (AlbericDeLajarte) [inline rouskinhf/list_datapoints.py:253]: I don't understand how this removes all datapoints if the structures/dms/shape are different

**Pull request closed without merging**